src/refmterr/lib/terminalEntry.re

```diff
@@ -19,25 +19,40 @@ let parseFromStdin =
   let reverseErrBuffer = {contents: []};
   let prettyPrintParsedResult =
     TerminalReporter.prettyPrintParsedResult(~refmttypePath, ~rawOutput);
+
+  let tryParseJson = line => Re.Pcre.pmatch(~rex=hasJsonError, line) ?
+    try (Some(Types_j.jsonResult_of_string(line))){
+      | e => None
+    } :
+    None;
   let forEachLine = line =>
     switch (
       reverseErrBuffer.contents,
       Re.Pcre.pmatch(~rex=fileR, line),
       Re.Pcre.pmatch(~rex=hasErrorOrWarningR, line),
       Re.Pcre.pmatch(~rex=hasIndentationR, line),
+      tryParseJson(line),
```

Comment:
I really don't like how this function (forEachLine) is implemented (and I definitely understand that you are just adding to it). Given that you just shortcut everything if you successfully parse some JSON and ignore it otherwise I think that you can add your logic without adding to this switch.

In general I think I would rather see the pattern matches extracted to variables and the collapsed cases extracted into either well named variables/variants or well named functions that are used with when clauses in the switch statement. The code in the last case (and most of the others) should also be extracted into functions IMO. It would be awesome if you picked this up given that you have spent some time with this code recently, but if not I think you should at least not shoehorn the JSON parsing into the switch.